hybrasyl/Objects/User.cs

```diff
@@ -1260,35 +1260,174 @@ public void SendVisibleGold(Gold gold)
         internal void UseSkill(byte slot)
         {
             var castable = SkillBook[slot];
-
-            Attack(castable);
-
-            Client.Enqueue(new ServerPacketStructures.Cooldown()
+            if (canCast(castable))
             {
-                Length = (uint)castable.Cooldown,
-                Pane = 1,
-                Slot = slot
-            }.Packet());;
+                bool spellCast = false;
+                foreach (var tar in PossibleTargets(castable))
+                {
+                    bool canAttack = !(tar is Merchant || (tar is User && (!(tar as User).Status.HasFlag(PlayerCondition.Pvp))));

```

Comment:
Npcs cannot be affect by damage at all, so this check is redundant. 

The attack method should handle all of the checks.  The original method that was here handled that case.

Reply:
Merchant check was in the original code, but easy enough to remove. 

The problem I had with the check being done in the attack method was that I couldn't verify if the attack actually happened, and thus whether to "charge" the user the cost. 

If I do this within the attack, then it would "charge" the cost for each target OR would charge once, but again, I couldn't guarantee that there was actually a target being attacked.

I'm happy to look at this again, if you really want the targeting done in the attack method.